test/base64_test.cpp

```diff
@@ -0,0 +1,170 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#include <migraphx/base64.hpp>
+#include "test.hpp"
+
+TEST_CASE(base64_encoding)
+{
+    std::string input_0    = "abc";
+    std::string expected_0 = "YWJj";
+    EXPECT(migraphx::b64_encode(input_0) == expected_0);
+
+    std::string input_1    = "abcd";
+    std::string expected_1 = "YWJjZA==";
+    EXPECT(migraphx::b64_encode(input_1) == expected_1);
+
+    std::string input_2    = "convolution";
+    std::string expected_2 = "Y29udm9sdXRpb24=";
+    EXPECT(migraphx::b64_encode(input_2) == expected_2);
+
+    std::string input_3    = "https://www.amd.com/en/products/software/rocm.html";
+    std::string expected_3 = "aHR0cHM6Ly93d3cuYW1kLmNvbS9lbi9wcm9kdWN0cy9zb2Z0d2FyZS9yb2NtLmh0bWw=";
+    EXPECT(migraphx::b64_encode(input_3) == expected_3);
+
+    std::string input_4    = "{1, 3, 7, 9}";
+    std::string expected_4 = "ezEsIDMsIDcsIDl9";
+    EXPECT(migraphx::b64_encode(input_4) == expected_4);
+}
+
+TEST_CASE(base64_RFC_test_vectors)
+{
+    std::string input_0    = "";
+    std::string expected_0 = "";
+    EXPECT(migraphx::b64_encode(input_0) == expected_0);
+
+    std::string input_1    = "f";
+    std::string expected_1 = "Zg==";
+    EXPECT(migraphx::b64_encode(input_1) == expected_1);
+
+    std::string input_2    = "fo";
+    std::string expected_2 = "Zm8=";
+    EXPECT(migraphx::b64_encode(input_2) == expected_2);
+
+    std::string input_3    = "foo";
+    std::string expected_3 = "Zm9v";
+    EXPECT(migraphx::b64_encode(input_3) == expected_3);
+
+    std::string input_4    = "foob";
+    std::string expected_4 = "Zm9vYg==";
+    EXPECT(migraphx::b64_encode(input_4) == expected_4);
+
+    std::string input_5    = "fooba";
+    std::string expected_5 = "Zm9vYmE=";
+    EXPECT(migraphx::b64_encode(input_5) == expected_5);
+
+    std::string input_6    = "foobar";
+    std::string expected_6 = "Zm9vYmFy";
+    EXPECT(migraphx::b64_encode(input_6) == expected_6);
+}
+
+// Following tests altered from
+// https://github.com/tobiaslocker/base64/blob/master/test/base64_tests.cpp
+TEST_CASE(base64_encodes_empty)
+{
+    std::string const expected{};
+    std::string const actual{migraphx::b64_encode({})};
+    EXPECT(expected == actual);
+}
+
+TEST_CASE(base64_encodes_three_bytes_zeros)
+{
+    std::array<std::uint8_t, 3> const input{0x00, 0x00, 0x00};
+    std::string expected{"AAAA"};
+    std::string actual{migraphx::b64_encode({input.begin(), input.end()})};
+    EXPECT(expected == actual);
+}
```

Comment:
Also the variables should be inlined: `EXPECT("AAAA" == migraphx::b64_encode({'\x00', '\x00', '\x00' })`.